apps/web-giddh/src/app/search/components/search-grid/search-grid.component.ts

```diff
@@ -503,4 +514,45 @@ export class SearchGridComponent implements OnInit, OnDestroy {
         });
         return queryForApi;
     }
+
+    /** Whether the number of selected elements matches the total number of rows. */
+    isAllSelected() {
+        const numSelected = this.selection.selected.length;
+        const numRows = this.dataSource.data.length;
+        return numSelected === numRows;
+    }
+
+    /** Selects all rows if they are not all selected; otherwise clear selection. */
+    masterToggle() {
```

Comment:
```suggestion
    selectAll() {
```
Please provide public class